### PR TITLE
[Merged by Bors] - Adapt to new discovery mechanism

### PR DESCRIPTION
**Pull request by maltesander**
## Description

- adapted to operator-rs 0.10.0
- compiled example yamls into one 

fixes https://github.com/stackabletech/nifi-operator/issues/194

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/pages/config_properties.adoc [hunk 1]
```diff
@@ -77,17 +74,10 @@ This setting is independent of the configured authentication method and will ove
...
-    chroot: /nifi
+  zookeeperConfigMapName: simple-nifi-znode
 ----
 NiFi in cluster mode requires a ZooKeeper ensemble for state management and leader election purposes, this operator at the moment does not support single node deployments without ZooKeeper, hence this is a required setting.
-Configuration happens via a ConfigMap, which needs to contain one key called `ZOOKEEPER` with the value being the ZooKeeper connection string.
-
-When using a Stackable provided ZooKeeper there is no need to manually create this ConfigMap, as it will be managed by the Stackable Operator for Apache ZooKeeper and kept up to date with a valid connection string for the target cluster.
-
-`chroot` is an optional parameter that allows keeping NiFi data under a specific znode in ZooKeeper, which is highly recommended to avoid any potential side effects of having multiple applications use the same ZooKeeper cluster.
+Configuration happens via a ConfigMap, which needs to contain two keys called `ZOOKEEPER_HOSTS` with the value being the ZooKeeper connection string and `ZOOKEEPER_CHROOT` with the value being the ZooKeeper chroot.
```
Comment (teozkr) [inline docs/modules/ROOT/pages/config_properties.adoc:80]: This is more of an implementation detail, the primary thing to note here should probably be that this typically comes from a `ZookeeperZnode`.

File: examples/simple-nificluster.yaml [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
+        config: {}
+---
+apiVersion: zookeeper.stackable.tech/v1alpha1
+kind: ZookeeperZnode
+metadata:
+  name: simple-nifi-znode
+spec:
+  clusterRef:
+    name: simple-zk
+    namespace: default
```
Comment (teozkr) [inline examples/simple-nificluster.yaml:26]:
```suggestion
```

`namespace` now defaults to the NS of the `ZookeeperZnode`.

File: docs/modules/ROOT/pages/config_properties.adoc (hunk 1 above)
Comment (maltesander) [reply docs/modules/ROOT/pages/config_properties.adoc]: added

**Review by teozkr — APPROVED** [review]

**Comment by maltesander** [comment]
bors merge

**Comment by lfrancke** [comment]
Sönke will be very proud of this build failure :)

**Pull request closed without merging**